CMakeLists.txt

```diff
@@ -56,7 +56,12 @@ set(${PROJECT_NAME}_HDRS
   include/class_loader/multi_library_class_loader.hpp
   include/class_loader/register_macro.hpp
 )
-add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS} ${${PROJECT_NAME}_HDRS})
+if(WIN32)
+  add_library(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SRCS} ${${PROJECT_NAME}_HDRS})
```

Comment:
@johnsonshih I would expect it to [fall back to SHARED](https://github.com/ros/catkin/blob/b1622e2042b8fa01df2896ea12d375acf6443beb/cmake/platform/windows.cmake#L7-L9) if BUILD_SHARED_LIBS is not specified. Did you experience a different behavior on Windows?

Reply:
On Windows, class_loader doesn't honor BUILD_SHARED_LIBS and is always built as DLL (it's specified in around line#69, defined CLASS_LOADER_BUILDING_DLL to build as DLL).
  I added this line to match the settings so it won't cause problem if BUILD_SHARED_LIBS is Off.


if(WIN32)

  # Causes the visibility macros to use dllexport rather than dllimport

  # which is appropriate when building the dll but not consuming it.

  target_compile_definitions(${PROJECT_NAME} PRIVATE "CLASS_LOADER_BUILDING_DLL")

endif()

Reply:
I'm not sure if that's a good idea.  The concern is the static data in class_loader, class_loader is a low level component used by a lot of components.  If class_loader is a static library, there will be multiple instances of class_loader static data within one process (that should be only one per process) that might cause unexpected side effect.  I will suggest keep it as is.